File: src/types/connection.ts
export interface ConnectionConfig {
  id: string;
  name: string;
  group: string;
  host: string;
  port: number;
  auth: AuthMethod;
  username: string;
  keepalive_interval: number;
  timeout: number;
  encoding: string;
  terminal_type: string;
  tags: string[];
  notes: string;
  created_at: string;
  updated_at: string;
  last_connected_at: string | null;
}

export type AuthMethod =
  | { Password: { password: string } }
  | { KeyFile: { key_path: string; passphrase: string | null } }
  | { Certificate: { cert_path: string; key_path: string } }
  | "KeyboardInteractive";

export function createDefaultConnection(): ConnectionConfig {
  const now = new Date().toISOString();
  return {
    id: crypto.randomUUID(),
    name: "",
    group: "default",
    host: "",
    port: 22,
    auth: { Password: { password: "" } },
    username: "root",
    keepalive_interval: 60,
    timeout: 10,
    encoding: "UTF-8",
    terminal_type: "xterm-256color",
    tags: [],
    notes: "",
    created_at: now,
    updated_at: now,
    last_connected_at: null,
  };
}

File: src-tauri/src/models/connection.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ConnectionConfig {
    pub id: String,
    pub name: String,
    pub group: String,
    pub host: String,
    pub port: u16,
    pub auth: AuthMethod,
    pub username: String,
    pub keepalive_interval: u32,
    pub timeout: u32,
    pub encoding: String,
    pub terminal_type: String,
    pub tags: Vec<String>,
    pub notes: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub last_connected_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum AuthMethod {
    Password { password: String },
    KeyFile { key_path: String, passphrase: Option<String> },
    Certificate { cert_path: String, key_path: String },
    KeyboardInteractive,
}

impl Default for ConnectionConfig {
    fn default() -> Self {
        let now = Utc::now();
        Self {
            id: uuid::Uuid::new_v4().to_string(),
            name: String::new(),
            group: String::from("default"),
            host: String::new(),
            port: 22,
            auth: AuthMethod::Password { password: String::new() },
            username: String::from("root"),
            keepalive_interval: 60,
            timeout: 10,
            encoding: String::from("UTF-8"),
            terminal_type: String::from("xterm-256color"),
            tags: Vec::new(),
            notes: String::new(),
            created_at: now,
            updated_at: now,
            last_connected_at: None,
        }
    }
}
